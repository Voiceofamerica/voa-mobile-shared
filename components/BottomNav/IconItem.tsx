
import * as React from 'react'
import { Subscription } from 'rxjs/Subscription'

import { iconItem } from './BottomNav.scss'

export type Props = React.Props<any> & {
  onClick?: () => void,
}

function IconItem ({ children, onClick = () => null }: Props) {
  return (
    <button className={iconItem} onClick={() => onClick()}>
      {
        children
      }
    </button>
  )
}

export default IconItem
