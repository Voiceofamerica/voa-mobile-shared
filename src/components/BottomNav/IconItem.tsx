
import * as React from 'react'

import { iconItem, activeItem } from './BottomNav.scss'

export type Props = React.Props<any> & {
  onClick?: () => void,
  className?: string,
  active?: boolean,
}

function IconItem ({ children, onClick = () => null, active = false, className = '' }: Props) {
  const iconClass = active
    ? `${iconItem} ${className} ${activeItem}`
    : `${iconItem} ${className}`

  return (
    <div className={iconClass} onClick={() => onClick()}>
      {
        children
      }
    </div>
  )
}

export default IconItem
