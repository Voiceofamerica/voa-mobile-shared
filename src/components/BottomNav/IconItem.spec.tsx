
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import IconItem from './IconItem'

describe(`<${IconItem.name} />`, () => {
  describe('shapshots', () => {
    it('should render empty', () => {
      const element = create((
        <IconItem />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given children', () => {
      const element = create((
        <IconItem>
          <div>child1</div>
          <div>child2</div>
        </IconItem>
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render active if prop set to true', () => {
      const element = create((
        <IconItem active />
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onClick', () => {
    it('should be called whenever the element is clicked', () => {
      const onClick = jest.fn()

      const item = shallow((
        <IconItem onClick={onClick} />
      ))

      item.simulate('click')

      expect(onClick).toHaveBeenCalledTimes(1)
    })

    it('should not throw an error if not provided and the element is clicked', () => {
      const item = shallow((
        <IconItem />
      ))

      item.simulate('click')
    })
  })
})
