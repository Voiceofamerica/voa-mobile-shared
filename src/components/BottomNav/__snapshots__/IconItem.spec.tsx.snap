// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconItem /> shapshots should render active if prop set to true 1`] = `
<button
  className="iconItem  activeItem"
  onClick={[Function]}
/>
`;

exports[`<IconItem /> shapshots should render empty 1`] = `
<button
  className="iconItem "
  onClick={[Function]}
/>
`;

exports[`<IconItem /> shapshots should render with the given children 1`] = `
<button
  className="iconItem "
  onClick={[Function]}
>
  <div>
    child1
  </div>
  <div>
    child2
  </div>
</button>
`;
